### PR TITLE
Reading telegram messages with circe

### LINKED ISSUE #3: Http Telegram API
Receive and send messages
Docs: https://core.telegram.org/bots

### PR DESCRIPTION
**Pull request by JTruji**
Fixes #3

### CODE REVIEW

File: src/main/scala/battery/bot/telegram/TelegramClient.scala [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    client
+      .expect[TelegramJSON](
+        telegramUri / s"bot$telegramToken" / "getUpdates"
+      )
+      .flatTap(hj => IO(println(hj)))
+//      .map(_.result.map(_.message.chat.id))
+
+//    client
+//      .expect[String](
+//        (telegramUri / s"bot$telegramToken" / "sendMessage")
```
Comment (AdrianRaFo) [inline src/main/scala/battery/bot/telegram/TelegramClient.scala:28]: this should be a separated non-commented method

File: src/main/scala/battery/bot/telegram/TelegramClient.scala [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+class TelegramClient(client: Client[IO], telegramToken: String) {
+val telegramUri = uri"https://api.telegram.org"
+  def callTelegram(): IO[TelegramJSON] = {
+
+//    client
+//      .expect[String](
+//        telegramUri / s"bot$telegramToken" / "getMe"
+//      )
+//      .map(hj => println(hj))
```
Comment (AdrianRaFo) [inline src/main/scala/battery/bot/telegram/TelegramClient.scala:17]: delete this

File: src/main/scala/battery/bot/Main.scala [hunk 3]
```diff
@@ -1,17 +1,18 @@
...
 
-object Main extends IOApp{
+object Main extends IOApp {
 
   override def run(args: List[String]): IO[ExitCode] = {
 
     val config = ConfigSource.default.loadOrThrow[Config]
-    println (config)
+    println(config)
     println(config.database)
```
Comment (AdrianRaFo) [inline src/main/scala/battery/bot/Main.scala:16]: these can be deleted now too

File: src/main/scala/battery/bot/Main.scala [hunk 4]
```diff
@@ -20,8 +21,8 @@ object Main extends IOApp{
 //      .load()
 //      .migrate
```
Comment (AdrianRaFo) [inline src/main/scala/battery/bot/Main.scala:22]: this should not be commented

File: src/main/scala/battery/bot/telegram/models/TelegramJSON.scala [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+import io.circe.generic.extras.semiauto.deriveConfiguredCodec
+
+case class TelegramJSON[A](
+    ok: Boolean,
+    description: Option[String],
+    result: List[A]
+)
+
+object TelegramJSON{
+  //implicit val codec: Codec[TelegramJSON] = deriveConfiguredCodec[TelegramJSON]
```
Comment (AdrianRaFo) [inline src/main/scala/battery/bot/telegram/models/TelegramJSON.scala:13]: delete this

File: src/main/scala/battery/bot/Main.scala [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+//      .migrate
+
+    val telegramClient = EmberClientBuilder.default[IO].build.map { client =>
+      new TelegramClient(client, config.telegramToken)
+    }
+
+    //telegramClient.use(_.telegramGetMe.as(ExitCode.Success))
+    telegramClient.use(_.telegramGetUpdate.as(ExitCode.Success))
+    //telegramClient.use(_.telegramGetMyCommands.as(ExitCode.Success))
+    //telegramClient.use(_.telegramDeleteMyCommands.as(ExitCode.Success))
```
Comment (AdrianRaFo) [inline src/main/scala/battery/bot/Main.scala:41]: these comments can be deleted

File: src/main/scala/battery/bot/Main.scala [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+    val ta = Transactor.fromDriverManager[IO](
+      config.database.driver,     // driver classname
+      config.database.url,     // connect URL (driver-specific)
+      config.database.user,                  // user
+      config.database.password               // password
+    )
+
+    val persistenceService = new PersistenceService(ta)
+    println(persistenceService.addPrice(1,0.123).unsafeRunSync())
```
Comment (AdrianRaFo) [inline src/main/scala/battery/bot/Main.scala:26]: this should be deleted(we never do unsafeRunSync)

File: src/main/scala/battery/bot/Main.scala (hunk 4 above)
Comment (AdrianRaFo) [reply src/main/scala/battery/bot/Main.scala:32]: still commented

File: src/main/scala/battery/bot/Main.scala (hunk 6 above)
Comment (JTruji) [reply src/main/scala/battery/bot/Main.scala:41]: I want to keep it because It could be useful in command`s branch. what's your point?

**Review by AdrianRaFo — APPROVED** [review]

**Pull request merged**